### ISSUE TITLE
Inconsistent behaviour of cabundle for webhooks and apiservices

### ISSUE THREAD

**Issue by olemarkus** [kubernetes/kubernetes#96799]
**What happened**:

Both `APIService` and webhookclientconfig (e.g `MutatingWebhookConfiguration`) has a `caBundle` field that states "If unspecified, system trust roots on the apiserver are used.".

For webhokclientconfig, this includes the cluster CA (used by kube-apiserver), for `APIService` it does not. If you have an apiregistration and you omit it, apiserver will complain about unknown CA. If you add the b64 of the cluster CA, it works.

**What you expected to happen**:

Expect that "system trust roots" is consistent across these two types. 

**How to reproduce it (as minimally and precisely as possible)**:

Basically do https://kubernetes.io/docs/tasks/extend-kubernetes/setup-extension-api-server/, but use the CA

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.19.3
- Cloud provider or hardware configuration: AWS
- OS (e.g: `cat /etc/os-release`): Ubuntu 20.04

- Kernel (e.g. `uname -a`): Linux #29-Ubuntu SMP Mon Oct 5 15:30:10 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
- Install tools: kops

**Comment by liggitt** [comment]
Are you pointing either object at the "kubernetes" service in the "default" namespace?

**Comment by olemarkus** [comment]
Both pointing to custom services in kube-system.

**Comment by liggitt** [comment]
I'm having trouble reproducing this... do you know how the API server is being invoked? Specifically, these options:

* is `--enable-aggregator-routing` set?
* is `--admission-control-config-file` specified, and does it include a mutating/validating admission config pointing at a kubeconfig file? if so, does that kubeconfig file include a CA bundle?

**Comment by olemarkus** [comment]
I tried to reproduce it again, and now I am getting unknown CA in both cases. I cannot say what is different from the last time I tried. 

`--enable-aggregator-routing` is set to false and there is no adminssion control config file set.

**Comment by olemarkus** [comment]
I really cannot reproduce this anymore. If there is no intention for kube-apiserver to trust its own CA, just close this one. Sorry for the inconvenience.

**Comment by liggitt** [comment]
> I really cannot reproduce this anymore. If there is no intention for kube-apiserver to trust its own CA, just close this one. Sorry for the inconvenience.

no worries, thanks for following up.

/close
